### PR TITLE
Adds diesel variant to 1836jr56

### PR DESCRIPTION
**Pull request by philcampeau**
Adds diesel variant to 1836jr56

### CODE REVIEW

File: lib/engine/game/g_1836_jr56/game.rb [hunk 1]
```diff
@@ -27,130 +27,164 @@ def cert_limit(_player = nil)
...
+            {
+              name: 'D',
+              on: 'D',
+              train_limit: 2,
+              tiles: %i[yellow green brown gray],
+              status: %w[fullcap facing_6 upgradable_towns no_loans],
+              operating_rounds: 3,
+            },
+          ]
+          phase_list.reject! { |p| p[:name] == 'D' } unless diesel_variant?
```
Comment (tobymao) [inline lib/engine/game/g_1836_jr56/game.rb:120]: change this to
diesel_variant? reject! : reject!

File: lib/engine/game/g_1836_jr56/game.rb [hunk 2]
```diff
@@ -27,130 +27,164 @@ def cert_limit(_player = nil)
...
+            {
+              name: 'D',
+              distance: 999,
+              price: 1100,
+              num: 22,
+              available_on: '6',
+              discount: { '4' => 350, "4'" => 350, '5' => 350, "5'" => 350, '6' => 350 },
+            },
+          ]
+          train_list.reject! { |t| t[:name] == 'D' } unless diesel_variant?
```
Comment (tobymao) [inline lib/engine/game/g_1836_jr56/game.rb:184]: same

File: lib/engine/game/g_1836_jr56/game.rb (hunk 1 above)
Comment (philcampeau) [reply lib/engine/game/g_1836_jr56/game.rb:120]: I'm not sure what you mean here. Could you elaborate?

Comment (philcampeau) [reply lib/engine/game/g_1836_jr56/game.rb:120]: @tobymao could you explain what you're requesting here? I'm not understanding.

**Pull request merged**